Source/src/core/GameObject.h

```diff
@@ -17,43 +17,54 @@
 
 namespace Hachiko
 {
-    class ComponentTransform;
-    class ComponentCamera;
-    class Program;
-    class Scene;
+class ComponentTransform;
+class ComponentCamera;
+class Program;
+class Scene;
 
-    class HACHIKO_API GameObject final : public ISerializable
-    {
-        friend class Component;
-
-    public:
-        GameObject(const char* name = "Unnamed");
-        GameObject(GameObject* parent,
-                   const float4x4& transform, 
-                   const char* name = "Unnamed", 
-                   UID uid = UUID::GenerateUID());
-        GameObject(GameObject* parent,
-                   const char* name = "Unnamed",
-                   UID uid = UUID::GenerateUID(),
-                   const float3& translation = float3::zero,
-                   const Quat& rotation = Quat::identity,
-                   const float3& scale = float3::one);
-        virtual ~GameObject();
+class HACHIKO_API GameObject final : public ISerializable
+{
+    friend class Component;
+
+public:
+    GameObject(const char* name = "Unnamed");
+    GameObject(GameObject* parent,
+                const float4x4& transform, 
+                const char* name = "Unnamed", 
+                UID uid = UUID::GenerateUID());
+    GameObject(GameObject* parent,
+                const char* name = "Unnamed",
+                UID uid = UUID::GenerateUID(),
+                const float3& translation = float3::zero,
+                const Quat& rotation = Quat::identity,
+                const float3& scale = float3::one);
+    virtual ~GameObject();
 
     void SetNewParent(GameObject* new_parent);
 
     void AddComponent(Component* component);
     bool AttemptRemoveComponent(Component* component);
-
-    Component* CreateComponent(Component::Type type);
     /// <summary>
-    /// Do not use this unless it's mandatory. Use AttemptRemoveComponent 
```

Comment:
Much wow, very needed.
Jokes aside, I tried breaking it creating 1200+ GO with 10 scripts at the same time creating objects and it worked wonders. Awesome job, as always.
Sonarcloud seems to be a bit annoyed by the fact that we use a pointer var to indicate _nullptr_, maybe changing all scene->root calls to directly _nullptr_ might be the solution to make it quiet? But at the same time it may create a less readable code... IDK, and besides, I dont fully understand how root works, is it _nullptr_, is it not? Not always?

Reply:
Hey, as we talked on discord, I'm gonna ignore that sonar bug as root is always non-nullptr and we can safely assume that. Thanks for your comments ❤️